subtests/docker_cli/dockerinspect/test_inspect_keys.py

```diff
@@ -0,0 +1,270 @@
+            divider = line.find(' |')
+            if divider >= 0:
+                lhs += line[:divider - 1] + "\n"
+                rhs += line[divider + 2:] + "\n"
+            else:
+                lhs += line + "\n"
+                rhs += line + "\n"
+
+        actual = inspect_keys.inspect_keys.filter_keys(mockself,
+                                                       json.loads(lhs))
+        self.maxDiff = None
+        self.assertEqual(actual, json.loads(rhs))
+
+    def test_basic(self):
+        """
+        Very simple test; makes sure that .Config.Labels is removed
+        """
+        self._run_one_test("""
+[
+{
+    "A": "B",
+    "Config": {
+       "Labels": { "a": [ "this gets deleted" ] },   | "Labels": {},
+       "Other":  [ "this is preserved" ]
+    }
+}
+]
+""")
+
+    def test_multi_net(self):
+        """
+        Multiple .NetworkSettings.Networks.xxx are renamed
+        """
+        self._run_one_test("""
+[
+{
+    "NetworkSettings": {
+        "Networks": {
+            "bridge": {                  | "RenamedForTesting_bridge": {
+                "EndpointID": "--preserved--"
+            },
+            "nextnet": {                 | "RenamedForTesting_nextnet": {
+                "AnotherPreservedKey": "yep"
+            },
+            "yetanothernet": {           | "RenamedForTesting_yetanothernet": {
+                "never": "mind"
+            },
+            "CamelCase": {               | "RenamedForTesting_CamelCase": {
+                "never": "mind"
+            }
+        }
+    }
+}
+]
+""")
+
+    def test_full(self):
+        """
+        Huge monster results, taken from a real-world instance.
+        """
+        self._run_one_test("""
+[
+{
+    "Id": "ed3b9325177ccd70717d58885dbeb53255b52f200be1dd7e65221e5eb805113e",
+    "Created": "2016-03-30T17:51:47.675978044Z",
+    "Path": "tail",
+    "Args": [
+        "-n1",
+        "/proc/1/cgroup"
+    ],
+    "State": {
+        "Status": "exited",
+        "Running": false,
+        "Paused": false,
+        "Restarting": false,
+        "OOMKilled": false,
+        "Dead": false,
+        "Pid": 0,
+        "ExitCode": 0,
+        "Error": "",
+        "StartedAt": "2016-03-30T17:51:49.006872458Z",
+        "FinishedAt": "2016-03-30T17:51:49.025916776Z"
+    },
+    "Image": "bf63a676257aeb7a....8f9b399004ef",
+    "ResolvConfPath": "/var/lib/docker/containers/..../resolv.conf",
+    "HostnamePath": "/var/lib/docker/containers/..../hostname",
+    "HostsPath": "/var/lib/docker/containers/..../hosts",
+    "LogPath": "/var/lib/docker/containers/..../....-json.log",
+    "Name": "/modest_colden",
+    "RestartCount": 0,
+    "Driver": "devicemapper",
+    "ExecDriver": "native-0.2",
+    "MountLabel": "system_u:object_r:svirt_sandbox_file_t:s0:c3,c28",
+    "ProcessLabel": "system_u:system_r:svirt_lxc_net_t:s0:c3,c28",
+        "AppArmorProfile": "",
+    "ExecIDs": null,
+    "HostConfig": {
+        "Binds": null,
```

Comment:
or encode it all inside a string - but then you can't easily use ``json_verify`` and ``json_reformat``.